packages/frontend-shared/cypress/support/mock-graphql/mountFragment.ts

```diff
@@ -134,15 +135,35 @@ export const registerMountFn = ({ plugins }: MountFnOptions = {}) => {
     }), mountingOptions).then(() => context)
   }
 
+  const mutationResolvers: Map<string, MutationResolver<any>> = new Map()
+
+  function setMutationResolver<Result, Variables, T extends TypedDocumentNode<Result, Variables>> (
+    document: T,
+    resolver: MutationResolver<T>,
+  ) {
+    const definition = document.definitions[0]
+
+    if (definition.kind === 'OperationDefinition' && definition.name) {
+      mutationResolvers[definition.name.value] = resolver
+    } else {
+      throw new Error('only use mutation documents in setMutationResolver first argument')
+    }
+  }
+
   Cypress.Commands.add('mountFragment', mountFragment)
 
+  Cypress.Commands.add('setMutationResolver', setMutationResolver)
+
   Cypress.Commands.add('mountFragmentList', (source, options) => {
     // @ts-expect-error - todo: tim fix
     return mountFragment(source, options, true)
   })
-}
 
-type ResultType<T> = T extends TypedDocumentNode<infer U, any> ? U : never
+  afterEach(() => {
+    // clean all resolvers after each test
+    mutationResolvers.clear()
+  })
```

Comment:
Should this be in a `beforeEach`? We generally recommend cleanup in `beforeEach` in our docs. Not sure there's a chance this doesn't get called, it might make no difference (if so ignore this comments).

Reply:
Done 70e298c